File: Algorithms/1512_Number_of_Good_Pairs/Solution.scala
object Solution {
  def numIdenticalPairs(nums: Array[Int]): Int = nums
    .map((_, 1))
    .groupMapReduce(_._1)(_._2)(_ + _)
    .map(kv => kv._2 * (kv._2 - 1) >> 1)
    .sum
}

File: Algorithms/1512_Number_of_Good_Pairs/solution.go
package main

func numIdenticalPairs(nums []int) int {
    freq := make(map[int]int)
    for _, n := range nums {
        freq[n]++
    }
    pairs := 0
    for _, v := range freq {
        pairs += v * (v - 1) >> 1
    }
    return pairs
}

func main() {

}
